elytron-common/pom.xml

```diff
@@ -28,4 +28,165 @@
     <artifactId>wildfly-elytron-integration-common</artifactId>
 
     <name>WildFly: Elytron Subsystem - Common Components</name>
+
+    <dependencyManagement>
+        <dependencies>
+            <dependency>
+                <groupId>org.wildfly.core</groupId>
+                <artifactId>wildfly-core-testbom</artifactId>
+                <version>${project.version}</version>
+                <type>pom</type>
+                <scope>import</scope>
+            </dependency>
+        </dependencies>
+    </dependencyManagement>
+
+    <dependencies>
+        <dependency>
+            <groupId>jakarta.authentication</groupId>
+            <artifactId>jakarta.authentication-api</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>org.jboss.logging</groupId>
+            <artifactId>jboss-logging-annotations</artifactId>
+            <scope>provided</scope>
+            <optional>true</optional>
+        </dependency>
+
+        <dependency>
+            <groupId>org.jboss.logging</groupId>
+            <artifactId>jboss-logging-processor</artifactId>
+            <scope>provided</scope>
+            <optional>true</optional>
+        </dependency>
+
+        <dependency>
+            <groupId>org.jboss.msc</groupId>
+            <artifactId>jboss-msc</artifactId>
+            <scope>provided</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>org.wildfly.security</groupId>
+            <artifactId>wildfly-elytron-realm-ldap</artifactId>
```

Comment:
Is this needed for elytron-common?

Reply:
Would be godo to check each dependency here to confirm it's needed.

Reply:
I'll take a second look. The dependencies I included were a subset of the original subsystem, so I might've reduced the number of dependencies by pulling out the wrong ones.

Reply:
Working through the Elytron dependencies, I've updated the list to this (pushed changes):

- `wildfly-elytron-auth-server`
- `wildfly-elytron-credential-store`
- `wildfly-elytron-keystore`
- `wildfly-elytron-ssl` (ideally will replace with `wildfly-elytron-ssl-common`)
- `wildfly-elytron-x500-cert-acme`
- with `org.wildfly.core:wildfly-server`

and then the transitive dependencies from this set:

- `wildfly-elytron-base` (from any module on the list)
- `wildfly-elytron-credential` (from `credential-store`)
- `wildfly-elytron-provider-util` (from `keystore`)
- `wildfly-elytron-security-manager` (from `wildfly-server`)
- `wildfly-elytron-x500-cert` and `wildfly-elytron-x500` (both from `x500-cert-acme`)

There are two main blockers to removing `wildfly-elytron-auth-server`:

1. `SecurityDomain` and `RealmMapper` are values that can be injected into a [server-ssl-context](https://github.com/cam-rod/wildfly-core/blob/194a812265cf11165c061572121f2c9fc631d392/elytron-common/src/main/java/org/wildfly/extension/elytron/common/SSLDefinitions.java#L1292-L1298). This one isn't too bad, I can move the implementation of the `TrivialAddHandler` outside of the common module.
2. The `CredentialSource` interface itself is part of the module. It ties back to the empty object [IdentityCredentials.NONE](https://github.com/wildfly-security/wildfly-elytron/blob/ab0080b71564640a889860f33ae462372e9e5d56/auth/server/base/src/main/java/org/wildfly/security/auth/server/IdentityCredentials.java#L404) (which is also directly used by [CredentialStoreUtility](https://github.com/cam-rod/wildfly-core/blob/194a812265cf11165c061572121f2c9fc631d392/elytron-common/src/test/java/org/wildfly/extension/elytron/common/CredentialStoreUtility.java#L92)). I don't see any obvious reason either of those classes need to be part of `auth-server`, but they both have to be moved to take out the dependency.

Reply:
Made similar changes in `wildfly-elytron-integration`, based on dependencies that aren't listed in the `wildfly-elytron-integration-common` POM. Listed Elytron dependencies:

- `wildfly-elytron-audit`
- `wildfly-elytron-auth-server` (since eventually we want to remove this) and `wildfly-elytron-auth-server-deprecated`
- `wildfly-elytron-auth-util`
- `wildfly-elytron-client`
- `wildfly-elytron-http-util`
- `wildfly-elytron-realm-ldap`, `-realm-jdbc`, and `-realm-token`
- `wildfly-elytron-sasl`
- `wildfly-elytron-ssl`

And transitive deps:

- `wildfly-elytron-asn1`, `-x500` and `-x500-cert` (from `-x500-cert-acme` via **common**)
- `wildfly-elytron-encryption` (from `-credential-store` via **common**)
- `wildfly-elytron-auth`,  `-base`, `-credential`, `-permission`, and `-util` (from `-auth-server-common`)
- `wildfly-elytron-mechanism-gssapi` and `-security-manager-action` (from `-auth-util`)
- `wildfly-elytron-http` (from `-http-util`)
- `wildfly-elytron-provider-util` (from `-client`)
- `wildfly-elytron-realm` (from any of the `-realm-*`)